src/components/sign-attributes/sign-attribute-modal.vue

```diff
@@ -0,0 +1,264 @@
+    private onAttributeChanged(
+        newAttribute: InterpretationAttributeDTO | null
+    ) {
+        this.setComment(newAttribute);
+    }
+
+    /*    private onAttributeValueChanged(newVal: ) {
+        for (const si of this.$state.artefactEditor
+    } */
+
+    private setComment(newAttribute: InterpretationAttributeDTO | null) {
+        this.comment = newAttribute?.commentary?.commentary || '';
+    }
+
+    private onCommentUpdated() {
+        if (this.isMultiSelect) {
+            console.warn(
+                "Can't update a comment when multiple signs are selected"
+            );
+            return;
+        }
+
+        const si = this.$state.artefactEditor.selectedSignsInterpretation[0]; // Only one element, since !isMultiSelect
+        const newAttr: InterpretationAttributeDTO = { ...this.attribute! };
+
+        newAttr.commentary = this.comment
+            ? ({ commentary: this.comment } as CommentaryDTO)
+            : undefined;
+
+        // Create an operation that will be added to the undo/redo management of the artefact editor
+        const op = new TextFragmentAttributeOperation(
+            si.id,
+            this.attribute!.attributeValueId,
+            newAttr
+        );
+
+        op.redo(); // Apply change
+        this.$state.eventBus.emit('new-operation', op);
+    }
+
+    private onDeleteAttribute() {
+        const ops: TextFragmentAttributeOperation[] = [];
+        for (const si of this.$state.artefactEditor
+            .selectedSignsInterpretation) {
+            const op = new TextFragmentAttributeOperation(
+                si.id,
+                this.attribute!.attributeValueId,
+                undefined
+            );
+            op.redo();
+            ops.push(op);
+        }
+        this.$state.eventBus.emit('new-bulk-operations', ops);
+        this.hide();
+    }
+
+    private onAttributeValueChanged(attrVal: AttributeValueDTO) {
+        const ops: TextFragmentAttributeOperation[] = [];
+        for (const si of this.$state.artefactEditor
+            .selectedSignsInterpretation) {
+            for (const attr of si.attributes.filter(
+                (a) => a.attributeValueId === this.attribute!.attributeValueId
+            )) {
+                const newAttr: InterpretationAttributeDTO = { ...attr };
+                newAttr.attributeValueId = attrVal.id;
+                newAttr.attributeValueString = attrVal.value;
+                this.$state.artefactEditor.selectedAttribute = newAttr;
+                const op = new TextFragmentAttributeOperation(
+                    si.id,
+                    attr.attributeValueId, // This is the old ID of the attribute
+                    newAttr
+                );
+                op.redo();
+                ops.push(op);
+            }
+        }
+        this.$state.eventBus.emit('new-bulk-operations', ops);
+        this.selected = null;
+    }
+
+    private hide() {
+        (this.$refs.signAttributeModalRef as any).hide();
```

Comment:
We do we need to cast to 'any'?  Is hide not a function on that object that Typescript can see?

Reply:
If we don't cast to any, hide doesn't exist.

Reply:
I did spend some time trying to figure out what the actual type was, but failed :-(

Reply:
Ah, I get it now. This comes from the bowels of some vue library. Don't worry about it then.